Sync/KeyBundle.swift

```diff
@@ -172,11 +172,29 @@ open class KeyBundle: Hashable {
     open func serializer<T: CleartextPayloadJSON>(_ f: @escaping (T) -> JSON) -> (Record<T>) -> JSON? {
         return { (record: Record<T>) -> JSON? in
             let json = f(record.payload)
-            let data = (json.rawString([writtingOptionsKeys.castNilToNSNull: true]) ?? "").utf8EncodedData
+            if json.isNull() {
```

Comment:
Quite possibly, yes. I wanted to limit the scope of this change, given the limited duration of testing — and FxA (and push?) tend not to encode arbitrary strings from the internet!

I left a note in the PR description or the bug comment suggesting that we look elsewhere, but I'd rather see that done as a follow-up.